src/lib/comms/sol-oic-client.c

```diff
@@ -169,6 +169,21 @@ sol_oic_resource_ref(struct sol_oic_resource *r)
     return r;
 }
 
+static void
+clear_vector_list(struct sol_vector *vector, void *data)
```

Comment:
What other approach? Take a look at the comment I did to @barbieri. if you have a better idea, I can do it.